Cox reusable action

alainamstutz opened the pull request:
Hi @venexia,
May I send you this PR? It's what we looked at together during the meeting - but one addition which we did not discuss:
- `cox_date_metfin_start_within6m` , see variable definition in `data_process.R`. 
- `landmark_date` is my input for the cox_start argument and is complete for all participants, defined as six months after type 2 diabetes diagnosis (`elig_date_t2dm`), the date of eligibility. Because of the landmark design, we disregard the first six months, only start follow-up then with everyone who is still alive and not LTFU at that point. 
- `cox_date_metfin_start_within6m` is only complete for those in intervention, i.e. started metformin mono during the first six months, but their exposure date should be reset to the landmark date. 
- Now, doing simply: cox_date_metfin_start_within6m = case_when(exp_bin_metfin_mono == TRUE ~ landmark_date, TRUE ~ as.Date(NA)) throws the following error: 

![Screenshot 2025-04-03 at 15 51 42](https://github.com/user-attachments/assets/1fd43d12-3a93-4785-a1a7-f94ffeb2c1e1)

Seems like an issue with some date overlap between landmark and exposure dates. If I add 1 day, it works - but probably not the cleanest workaround...if you have some insights regarding the error and suggestion for a better workaround, I'd be very glad! 

Thanks a lot!

venexia commented:
Hi @alainamstutz. I think this is happening because `cox-ipw` expects at least some pre-exposure events, whereas in your setup, everyone is effectively exposed instantaneously at the start of the study. I think the best solution for you is to set `cox-start = landmark - 1 day` as this will ensure no events happen in the pre-exposure episode and it preserves the time between your exposure and outcome date. If you can wait, I can probably do a quick PR to cox-ipw to resolve this next week.

alainamstutz commented:
I see - no worries, I'll wait, no rush!
Thank you very much!

venexia commented:
Good news: `cox-ipw::v0.0.35` has been released and supports cases with no pre-exposure events!

alainamstutz commented:
Thanks, @venexia , great news! 
I am really sorry, I had a another bug fix suggestion for cox-ipw, and had it all written up in this PR, but I now realise that it wasn't saved - was during my train travel journey, probably bad internet at the time of submission :-( Anyway, let me summarise again: 

After our discussion of how to best integrate the stratification variable in all analyses, I went back to the dataset definition and renamed `cov_cat_region` to `strat_cat_region`. I thought that's the cleanest way and whenever I call in the covariates in the models, I can simply use `cov_`. However, this resulted in a bug on this branch, saying that I didn't feed in a correct stratification variable. I went to the log files and realized that cox-ipw automatically creates `strat_cat_region` from `cov_cat_region` during the process and, hence, there were 2 variables with the same name and it renamed my stratification variable to `strat_cat_region.y` and resulted in an error. Now, of course, this is quite a coincidence that we chose the same variable names - but I can imagine that others maybe also feed in a stratification variable with the prefix `strat_`. I renamed `strat_cat_region` back to `cov_cat_region` and it works fine again (see last commit), but obviously that's not my preferred option... What do you think? Sorry again for not double-checking that my previous message went through!

venexia commented:
Hi @alainamstutz - no problem, I can take a look. Did you identify where the renaming happens by any chance?

alainamstutz commented:
@venexia , 
I think it's in [1] "Add strata information to data"
At least there, for the first time, I see `strat_cat_region.x` and `strat_cat_region.y`
In [1] "Add indicators for episode" summary output it is still `strat_cat_region`, then during the following no summary output of variables:
[1] "Calculate events in each time period"
[1] "Calculate number of events per episode"
[1] "Add number of events to episode info table"
[1] "Calculate person-time in each episode"
[1] "Calculate median person-time"
[1] "Add time from previous episodes to median person-time"
until [1] "Add strata information to data"

venexia commented:
Good news! cox-ipw:v0.0.36 has been released so hopefully the strata issue will be resolved.

alainamstutz commented:
thanks, it's working beautifully now !

alainamstutz commented:
hi, @venexia, is this PR ok from your side - would you approve it?

venexia reviewed (approved):
Hi @alainamstutz - sorry for the delay, all good!
